src/Quibill.Domain/Models/ITransaction.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Quibill.Domain
+{
+    internal interface ITransaction
+    {
+        int TransactionId { get; }
+        float TransactionAmount { get;}
+        string TransactionType { get; } // TODO make this an enum?
+        System.DateTime TransactionDate { get; }
```

Comment:
I was probably unclear about this, this DateTime was just supposed to store the Date the User adds the transaction, not necessarily the date the transaction is applied to their balance. Maybe just rename it AddDate?